pkg/controllers/gateway/gateway_controller.go

```diff
@@ -94,12 +96,34 @@ func (r *GatewayReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ct
 		}
 		return ctrl.Result{}, nil
 	}
+	if !controllerutil.ContainsFinalizer(previous, GatewayFinalizer) {
+		controllerutil.AddFinalizer(previous, GatewayFinalizer)
+		err = r.Update(ctx, previous)
```

Comment:
are we updating here to ensure we always add the finalizer?

Reply:
yes

Reply:
shouldn't this if statement also check that the resource is not deleting (i.e. don't re-add the finalizer when we want it removed)?

Reply:
added check for the deletion timestamp - should prevent the addition of the finalizer if we are deleting the gateway. And requeued (is it even a word? :thinking: ) it